temporal-sdk/src/test/java/io/temporal/workflow/activityTests/ActivityThrowingErrorTest.java

```diff
@@ -59,22 +67,45 @@ public void activityThrowsError() {
                     .setInitialInterval(Duration.ofMinutes(2))
                     .build())
             .build();
+  }
+
+  @Test
+  public void activityThrowsError() {
+    String name = testName.getMethodName();
+    WorkflowClient client = testWorkflowRule.getWorkflowClient();
+    TestWorkflow3 workflow = client.newWorkflowStub(TestWorkflow3.class, options);
 
-    TestWorkflow1 workflow = client.newWorkflowStub(TestWorkflow1.class, options);
     try {
-      workflow.execute(UUID.randomUUID().toString());
+      workflow.execute(name, 1);
     } catch (WorkflowFailedException e) {
       assertTrue(e.getCause() instanceof ActivityFailure);
       assertTrue(e.getCause().getCause() instanceof ApplicationFailure);
-      assertEquals("java.lang.Error", ((ApplicationFailure) e.getCause().getCause()).getType());
+      assertEquals("fail", ((ApplicationFailure) e.getCause().getCause()).getType());
```

Comment:
let's put this "fail" in a constant so it's dead-clear where this value is defined and where it comes from.